lib/src/facts/openbsd/processor_resolver.cc

```diff
@@ -0,0 +1,67 @@
+#include <internal/facts/openbsd/processor_resolver.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <sys/types.h>
+#include <sys/sysctl.h>
+
+using namespace std;
+
+namespace facter { namespace facts { namespace openbsd {
+
+    processor_resolver::data processor_resolver::collect_data(collection& facts)
+    {
+        auto result = posix::processor_resolver::collect_data(facts);
+        size_t len;
+        int mib[2];
+        mib[0] = CTL_HW;
+
+        // Get the physical count of processors
+        len = sizeof(result.logical_count);
+        mib[1] = HW_NCPUFOUND;
```

Comment:
That was a pasto in the code; `HW_NCPUFOUND` should be the physical number of processors (it's the number of processors/cores actually present in the machine). `HW_NCPU` should the number of logical CPUs (it's the number of processors/cores actually in use).
Does this mapping make sense?